### PR TITLE
Adding `db.Config.Application` field to set the `application_name` configuration parameter

### PR DESCRIPTION
**Pull request by dhermes**
## PR Summary

 - **Type:** Features
 - **Intended Change Level:** minor

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: db/config.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ func NewConfigFromDSN(dsn string) (*Config, error) {
 			config.SSLMode = strings.TrimPrefix(piece, "sslmode=")
 		} else if strings.HasPrefix(piece, "search_path=") {
 			config.Schema = strings.TrimPrefix(piece, "search_path=")
+		} else if strings.HasPrefix(piece, "application_name=") {
```
Comment (dhermes) [inline db/config.go:41]: See:

- https://www.postgresql.org/docs/12/runtime-config-logging.html#GUC-APPLICATION-NAME
- https://www.datadoghq.com/blog/collect-postgresql-data-with-datadog/#collecting-custom-postgresql-metrics-with-datadog

Comment (dhermes) [reply db/config.go:41]: I plan on using a Kubernetes pod name as the `application_name` to uniquely identify connection count (and Idle / Active state) within Datadog monitors

File: db/config.go [hunk 2]
```diff
@@ -61,22 +63,23 @@ func NewConfigFromDSN(dsn string) (*Config, error) {
...
-//	-	DB_MAX_CONNECTIONS   = MaxConnections
-//	-	DB_MAX_LIFETIME      = MaxLifetime
-//	-	DB_BUFFER_POOL_SIZE  = BufferPoolSize
+//  -  DB_ENGINE            = Engine
+//  -  DATABASE_URL         = DSN     //note that this has precedence over other vars (!!)
+//  -  DB_HOST              = Host
+//  -  DB_PORT              = Port
+//  -  DB_NAME              = Database
+//  -  DB_SCHEMA            = Schema
+//  -  DB_APPLICATION_NAME  = ApplicationName
```
Comment (wcharczuk) [inline db/config.go:72]: `Application`? the `Name` feels redundant, what do you think?

Comment (wcharczuk) [reply db/config.go:72]: Strictly speaking, that is the name in the underlying driver dsn so i'm fine with it.

**Review by wcharczuk — APPROVED** [review]
LGTM, plz port to golang.blend.com

File: db/config.go (hunk 1 above)
Comment (wcharczuk) [reply db/config.go:41]: yeah makes sense

File: db/config.go (hunk 2 above)
Comment (dhermes) [reply db/config.go:72]: I'm OK with that. I chose it because the `libpq` parameter is `application_name`

File: db/config.go (hunk 1 above)
Comment (wcharczuk) [reply db/config.go:41]: is this field supported in postgres < 12?

**Comment by dhermes** [comment]
> LGTM, plz port to golang.blend.com

Totally! (Definitely was planning on it)

File: db/config.go (hunk 2 above)
Comment (wcharczuk) [reply db/config.go:72]: yeah lets keep it at `application_name` for now

File: db/config.go (hunk 1 above)
Comment (dhermes) [reply db/config.go:41]: - https://www.postgresql.org/docs/9.5/runtime-config-logging.html#GUC-APPLICATION-NAME
- https://www.postgresql.org/docs/9.6/runtime-config-logging.html#GUC-APPLICATION-NAME
- https://www.postgresql.org/docs/10/runtime-config-logging.html#GUC-APPLICATION-NAME
- https://www.postgresql.org/docs/11/runtime-config-logging.html#GUC-APPLICATION-NAME

AFAICT it was introduced in 9.0, but earlier than 9.5 is completely unsupported so we're all good with 9.5+

File: db/config.go (hunk 2 above)
Comment (dhermes) [reply db/config.go:72]: SGTM

**Pull request merged**